### ISSUE TITLE
response()->json utf8 issues

### ISSUE THREAD

**Issue by sandrodz** [skipperbent/simple-php-router#333]
It returns gibberish (see name):

`[{"trans_id":"a0N1lXIfjome+JJOUXK8A4fvQEs=","status":"created","name":null,"for":null},{"trans_id":"eT0TawffDyArKiT8JtB0RiKa4Jg=","status":"created","name":null,"for":null},{"trans_id":"xvw\/GoksovyxSvftSRHBgTjc8pE=","status":"created","name":null,"for":null},{"trans_id":"+pJ2pSPm5zJanlMnpLIllAr5biM=","status":"created","name":null,"for":null},{"trans_id":"k4ribuEu0E8RWlEMla+YQh03org=","status":"created","name":"\u10e1\u10d0\u10dc\u10d3\u10e0\u10dd","for":null},{"trans_id":"54hj7Oqvm1L3al8mtpQzzPBkMy8=","status":"success","name":"\u10d5\u10d0\u10ee\u10e2\u10d0\u10dc\u10d2 \u10db\u10d4\u10d2\u10e0\u10d4\u10da\u10d8\u10e8\u10d5\u10d8\u10da\u10d8","for":null},{"trans_id":"igd\/UEoLdc9d\/t9TZ+YkYkO1Chc=","status":"failed","name":"\u10e1\u10ee\u10d5\u10d0 \u10e0\u10d0\u10db\u10d4","for":null},{"trans_id":"GK0nResZI23FrWU3sVnRL05v9zA=","status":"success","name":"\u10d5\u10d8\u10dc \u10d5\u10d8\u10dc \u10d0\u10e0\u10d8\u10e1 \u10e1\u10d0\u10e5\u10d0\u10e0\u10d7\u10d5\u10d4\u10da\u10dd\u10e8\u10d8","for":null},{"trans_id":"i5YVfAyzmy\/JtCpHiq7cquOOGZU=","status":"created","name":null,"for":null}]`

Simple variable dump, not going throng this method:

`array(9) { [0]=> array(4) { ["trans_id"]=> string(28) "a0N1lXIfjome+JJOUXK8A4fvQEs=" ["status"]=> string(7) "created" ["name"]=> NULL ["for"]=> NULL } [1]=> array(4) { ["trans_id"]=> string(28) "eT0TawffDyArKiT8JtB0RiKa4Jg=" ["status"]=> string(7) "created" ["name"]=> NULL ["for"]=> NULL } [2]=> array(4) { ["trans_id"]=> string(28) "xvw/GoksovyxSvftSRHBgTjc8pE=" ["status"]=> string(7) "created" ["name"]=> NULL ["for"]=> NULL } [3]=> array(4) { ["trans_id"]=> string(28) "+pJ2pSPm5zJanlMnpLIllAr5biM=" ["status"]=> string(7) "created" ["name"]=> NULL ["for"]=> NULL } [4]=> array(4) { ["trans_id"]=> string(28) "k4ribuEu0E8RWlEMla+YQh03org=" ["status"]=> string(7) "created" ["name"]=> string(18) "სანდრო" ["for"]=> NULL } [5]=> array(4) { ["trans_id"]=> string(28) "54hj7Oqvm1L3al8mtpQzzPBkMy8=" ["status"]=> string(7) "success" ["name"]=> string(58) "ვახტანგ მეგრელიშვილი" ["for"]=> NULL } [6]=> array(4) { ["trans_id"]=> string(28) "igd/UEoLdc9d/t9TZ+YkYkO1Chc=" ["status"]=> string(6) "failed" ["name"]=> string(25) "სხვა რამე" ["for"]=> NULL } [7]=> array(4) { ["trans_id"]=> string(28) "GK0nResZI23FrWU3sVnRL05v9zA=" ["status"]=> string(7) "success" ["name"]=> string(69) "ვინ ვინ არის საქართველოში" ["for"]=> NULL } [8]=> array(4) { ["trans_id"]=> string(28) "i5YVfAyzmy/JtCpHiq7cquOOGZU=" ["status"]=> string(7) "created" ["name"]=> NULL ["for"]=> NULL } }`

**Comment by skipperbent** [comment]
Hi @sandrodz.

Thanks for taking the time to report this issue.

In version 3.4.8.0 I've added some extra parameters to the `json` method in the `Response` class, including an `$options` parameter that will allow you to pass json-options. 

Setting the `JSON_UNESCAPED_UNICODE` constant on the `$options` parameter will disable hexadecimal conversion for the `json_encode` function.

More info on available options here:
http://php.net/manual/en/function.json-encode.php

**Example:**

```php
return response()->json($data, JSON_UNESCAPED_UNICODE);
```

/ Simon

**Comment by sandrodz** [comment]
Thanks for the quick fix :)

I'm still getting this:
> [{"trans_id":"6pedrakEVMPtKPssIqL1sh0H5mM=","status":"created","name":"4125706003904921","for":null,"note":null,"created_at":"2017-12-06 15:19:06"},{"trans_id":"ZOk63VkBHw2+0hQXvvjDv\/bARBo=","status":"created","name":"áƒ¢áƒ”áƒ¡áƒ¢áƒ”áƒ ","for":null,"note":null,"created_at":"2017-12-06 15:19:20"},{"trans_id":"I0cPUW\/PCE+ZoAf\/3vGFhHRhtcY=","status":"created","name":"áƒ¢áƒ”áƒ¡áƒ¢áƒ”áƒ ","for":null,"note":null,"created_at":"2017-12-06 15:20:05"},{"trans_id":"l8ah+gLawPMLntjDxi8NEAmoIkg=","status":"success","name":"áƒ¢áƒ”áƒ¡áƒ¢áƒ”áƒ ","for":null,"note":null,"created_at":"2017-12-06 15:20:17"}]

while directly doing this `echo json_encode($orders, JSON_UNESCAPED_UNICODE);` works

> [{"trans_id":"6pedrakEVMPtKPssIqL1sh0H5mM=","status":"created","name":"4125706003904921","for":null,"note":null,"created_at":"2017-12-06 15:19:06"},{"trans_id":"ZOk63VkBHw2+0hQXvvjDv\/bARBo=","status":"created","name":"ტესტერ","for":null,"note":null,"created_at":"2017-12-06 15:19:20"},{"trans_id":"I0cPUW\/PCE+ZoAf\/3vGFhHRhtcY=","status":"created","name":"ტესტერ","for":null,"note":null,"created_at":"2017-12-06 15:20:05"},{"trans_id":"l8ah+gLawPMLntjDxi8NEAmoIkg=","status":"success","name":"ტესტერ","for":null,"note":null,"created_at":"2017-12-06 15:20:17"}]

**Comment by skipperbent** [comment]
@sandrodz 

Hmm that's weird- because `response()->json` is just setting the header before calling `json_encode` - exactly like your working example, so you should get the same response.

Judging from the non-working response, it looks like the hexadecimal conversion has been disabled, but that the output for some reason is being encoded. 

Try editing the `response()->json()` and replace line 98 with the code below.  

```php
$this->header('Content-Type: application/json; charset=utf-8');
```

/ Simon

**Comment by sandrodz** [comment]
Yep, that helped. Can you push a version? it feels dirty to have modified file in vendors dir :)

**Comment by skipperbent** [comment]
@sandrodz It was just to confirm that it worked with your data :)

The fix is available in version 3.4.9.0.

**Comment by sandrodz** [comment]
Thanks!
